### PR TITLE
feat(idea-comment): add comment on idea

### PR DESCRIPTION
**Pull request by yofriadi**
review guys @akbarsahata @ibnuimam @azhariemuhammad thankyou

### Changelog
- add create comment on idea

### Insomnia
https://api.jsonbin.io/b/5ff489cc09f7c73f1b6e00fa/1

### CODE REVIEW

**Review by azhariemuhammad — APPROVED** [review]

File: src/api/controllers/ideaComment.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        response: 204,
+      },
+      onRequest: authenticate,
+    },
+  })
+  async commentIdea(
+    req: AuthenticatedRequest<{
+      Params: {id: string};
+      Body: {comment: string};
+      User: Record<string, unknown>;
```
Comment (akbarsahata) [inline src/api/controllers/ideaComment.ts:32]: User ga perlu di define pake `Record<string, unknown>` lagi, karena `req.user.user` dan `req.user.profile` udah didefine type-nya

File: src/api/entity/ideaComment.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @UpdateDateColumn({type: 'timestamp', default: () => 'NOW()'})
+  updatedAt: Date;
+
+  @DeleteDateColumn({type: 'timestamp', nullable: true})
+  deletedAt: Date;
+
+  @Column({type: 'text'})
+  comment: string;
+
+  @ManyToOne(() => Idea, (idea: Idea) => idea.likes)
```
Comment (akbarsahata) [inline src/api/entity/ideaComment.ts:29]: ini bukannya `(idea) => idea.comments` ya?

**Review by ibnuimam — APPROVED** [review]

File: src/api/entity/ideaComment.ts (hunk 2 above)
Comment (ibnuimam) [reply src/api/entity/ideaComment.ts:29]: @yofriadi di-resolve dulu ya.. bener harusnya kan idea.commnets no like in idea commnets, like cuma ada di idea.

Comment (yofriadi) [reply src/api/entity/ideaComment.ts:29]: oiya okee, nice catch thankyou

**Pull request merged**